### PR TITLE
[Front-End] Add seo guide

### PR DESCRIPTION
**Pull request by emanuelcasco**
# Summary

Add seo guide to frontend documentation.

### CODE REVIEW

File: frontend/docs/seo-guide.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+- Is concise and **contain your best keywords**.
+
+- Each page should have its own **exclusive title**.
+
+- Here you can find an (snippet optimizer)[http://www.seomofo.com/snippet-optimizer.html] that allows you to see how your titles look on Google and other search results.
+
+_Why:_
+> Title tags are a major factor in helping search engines understand what your page is about, and they are the first impression many people have of your page. [Read more...](https://moz.com/learn/seo/title-tag)
+
```
[the remark this answers is not in the document]
Comment (FrankIglesias) [reply frontend/docs/seo-guide.md]: `<title>Title of the document</title>`

File: frontend/docs/seo-guide.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+#### Images
+
```
Comment (FrankIglesias) [inline frontend/docs/seo-guide.md]: we can add some image optimizers like:
https://tinypng.com/
https://jakearchibald.github.io/svgomg/

File: frontend/docs/seo-guide.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+Almost every content has to have any image in it. But, how this affect SEO? Well, without the proper optimization, however, you’re wasting a valuable SEO asset. Plus, the larger the file size, the longer it takes your page to load, and the higher your bounce rate will be.
+
+- Be sure that your images are no larger than they need to be, that they are in the right file format.
+  _Why:_
+  > Your pages load time will vary based on the number and size of your images.
+
+- All images must have an `alt` and `title` attributes implemented that describes its content.
+  _Why:_
+  > ALT text attaches a description to your pictures so that they show up in Google and other search engine's image results. [Read more...](https://yoast.com/image-seo-alt-tag-and-title-tag-optimization/)
+
```
Comment (FrankIglesias) [inline frontend/docs/seo-guide.md]: Another important thing is to specify when to use an SVG or an PNG. When the image is an icon not a picture of something. SVG is the best choice

File: frontend/docs/seo-guide.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+  ```
+
+_Why:_
+> Leaving this out could impact how your page renders in the browser.
+
+#### Meta descriptions
+
+It explains to search engines and searchers themselves what your page is about. Describes the page to searchers as they read through the SERPs.
+
+- Should be 160-300 characters long. Keep in mind that the "optimal" length will vary depending on the situation, and your primary goal should be to provide value and drive clicks.
```
Comment (FrankIglesias) [inline frontend/docs/seo-guide.md]: *160  - 300 characters long *

File: frontend/docs/seo-guide.md [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+
+#### Images
+
+Almost every content has to have any image in it. But, how this affect SEO? Well, without the proper optimization, however, you’re wasting a valuable SEO asset. Plus, the larger the file size, the longer it takes your page to load, and the higher your bounce rate will be.
+
+- Be sure that your images are no larger than they need to be, that they are in the right file format.
+  _Why:_
+  > Your pages load time will vary based on the number and size of your images.
+
+- All images must have an `alt` and `title` attributes implemented that describes its content.
```
Comment (FrankIglesias) [inline frontend/docs/seo-guide.md]: are titles necessary?

**Comment by emanuelcasco** [comment]
@FrankIglesias can you review this again?

File: frontend/docs/seo-guide.md (hunk 3 above)
Comment (FrankIglesias) [reply frontend/docs/seo-guide.md]: :warning: image is a bad example for alts (also not recommended for SEO, as it does not represent the content

File: frontend/docs/seo-guide.md [hunk 6]
```diff
@@ -8,54 +8,71 @@ The aim of this document is to show good practices and conventions to improve yo
...
+
+**Search Engine Marketing (SEM)** is a form of Internet marketing that involved the promotion of websites by increasing their visibility in search engine results pages (SERPS) through optimization and advertising.
+
+#### SEO
+
+**Search Engine Optimization (SEM)** is a technical process through which differents changes in your websitie structure and content are implemented in order to improve its visibility en organic results from search engines.
+
+To know more please read [Google 101: How Google crawls, indexes and serves the web](https://support.google.com/webmasters/answer/70897)
+
+#### PPC
```
Comment (FrankIglesias) [inline frontend/docs/seo-guide.md]: :+1:

File: frontend/docs/seo-guide.md (hunk 3 above)
Comment (FrankIglesias) [reply frontend/docs/seo-guide.md]: @emanuelcasco

File: frontend/docs/seo-guide.md [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+- [HTML Validator](https://validator.w3.org/).
+- [CSS Validator](https://jigsaw.w3.org/css-validator/).
+
+Keep it in mind, W3C validation is not the “SEO magic bullet” to top rankings.
+
+#### Pagination
+
+Pagination is used to improve user experience. Instead of serving a lengthy page of results, it presents several smaller pieces (and arguably the most relevant first) that are easier to consume. Another advantage is that shorter pages load faster. But it can sometimes cause problems for search engines, if it is not correctly implemented.
+
+Implement the `rel="next"` and `rel="prev"` tags. We can use these elements to specify a relationship between multiple URLs.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: `rel` is an attribute, not a tag

File: frontend/docs/seo-guide.md [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+
+#### Pagination
+
+Pagination is used to improve user experience. Instead of serving a lengthy page of results, it presents several smaller pieces (and arguably the most relevant first) that are easier to consume. Another advantage is that shorter pages load faster. But it can sometimes cause problems for search engines, if it is not correctly implemented.
+
+Implement the `rel="next"` and `rel="prev"` tags. We can use these elements to specify a relationship between multiple URLs. 
+
+[Read more...](https://webdesign.tutsplus.com/articles/helping-search-engines-handle-pagination--webdesign-10035)
+
+### 2.5 Accesability
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: Accessibility haha

File: frontend/docs/seo-guide.md [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+      - [Meta content type](#meta-content-type)
+      - [Meta descriptions](#meta-descriptions)
+      - [Viewport](#viewport)
+      - [Language](#language)
+      - [Keywords](#keywords)
+   4. [Structure](#2.4-structure)
+      - [Headings](#headings)
+      - [Write valid code](#write-valid-code)
+      - [Pagination](#pagination)
+   5. [Accesability](#2.5-accesability)
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: Accessibility

File: frontend/docs/seo-guide.md [hunk 10]
```diff
@@ -0,0 +1,393 @@
...
+[Read more...](https://webdesign.tutsplus.com/articles/helping-search-engines-handle-pagination--webdesign-10035)
+
+### 2.5 Accesability
+
+#### Status code errors
+
+- **1XX:** These are informational codes and they indicate a provisional response, you should very rarely explicitly send a 1xx response.
+- **2XX:** These are success messages, this is what will be returned by default when a web page is successfully served up.
+- **3XX:** These are all redirection messages. This means that the client needs to perform an additional action in order to correctly complete the request. For more information go to [Reditections section](#redirections).
+- **4XX:** These are all client error messages. The most well known of these is the 404 message, but there are others, which I will get into in more details later.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: `in more detail later`

File: frontend/docs/seo-guide.md [hunk 11]
```diff
@@ -0,0 +1,393 @@
...
+  - **Provide a 403 error page** (access denied).
+  - **Provide a 404 error page** (Page not found).
+- **5XX:** These are all server error messages. Ensure your page does not return 500 errors.
+
+#### Sitemap
+
+I should start with this, Google does not index your pages because you asked for it. Google indexes pages because it founds them and considers them good enough quality to be indexed.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: `them good enough to be indexed`

File: frontend/docs/seo-guide.md [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+#### Sitemap
+
+I should start with this, Google does not index your pages because you asked for it. Google indexes pages because it founds them and considers them good enough quality to be indexed. 
+
+Sitemaps are machine readable URL submission lists in various formats, e.g., XML or plain text. It’s like an archive of every webpage in your website. **Use your sitemaps as sleuthing tools to discover and eliminate indexation problems**, and only let/ask Google to index the pages you know Google is going to want to index.
+
+Sitemaps are part of Robots Exclusion Protocol (REP), which we will cover in some lines.
+
+- If you’ve got a big site, use dynamic sitemaps. You can use plugins for this, for example: [DynamicSitemaps (RoR)](https://github.com/lassebunk/dynamic_sitemaps).
+- Not include pages in your sitemap that are blocked by robots.txt or have set meta robots `'noindex,follow'`.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: `Don't include pages`

File: frontend/docs/seo-guide.md [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+#### Canonical Tag
+Canonical URL **allows you to tell search engines that certain similar URLs are actually one and the same**. Sometimes you have products or content that is accessible under multiple URLs, or even on multiple websites. Using a canonical URL (an HTML link tag with attribute rel=canonical) these can exist without harming your rankings.
+
+You should use www or not in your web site’s canonical hostname? For SEO **it really doesn’t matter at all as long as you choose one**. For example:
+
+- https://www.my-page.com
+
+- https://my-page.com
+
+What do tou have to do the set the canonical URL?
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: What do **you** have to do **to** set

File: frontend/docs/seo-guide.md [hunk 14]
```diff
@@ -0,0 +1,393 @@
...
+_Why:_
+
+> This method help us to prevent problems caused by identical or "duplicate" content appearing on multiple URLs. [Read more...](https://moz.com/learn/seo/canonicalization)
+
+#### URLs
+
+Search engines and visitors can interact with your website more effectively when you implement URLs that describe your pages content.
+
+  - URLs should not be grater than 100 charaters.
+  - Avoid use more than 3 levels in your URLs. What I mean with level? https://yourdomain.com/level1/level2/level3.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: Avoid using more

File: frontend/docs/seo-guide.md [hunk 15]
```diff
@@ -0,0 +1,393 @@
...
+  - URLs must be user-friendly. An URL have to describe page content and include keyword in it. For example:
+    - BAD: https://yourdomain.com/123123123123lkjlkjlkjlkj.
+    - GOOD: https://yourdomain.com/how-to-implement-good-seo-strategy.
+- There are no params in the URL, this can be setup in Google Search Console.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: this is not indented and i don't understand what you mean here

File: frontend/docs/seo-guide.md [hunk 16]
```diff
@@ -0,0 +1,393 @@
...
+
+  - URLs should not be grater than 100 charaters.
+  - Avoid use more than 3 levels in your URLs. What I mean with level? https://yourdomain.com/level1/level2/level3.
+  - URLs must be user-friendly. An URL have to describe page content and include keyword in it. For example:
+    - BAD: https://yourdomain.com/123123123123lkjlkjlkjlkj.
+    - GOOD: https://yourdomain.com/how-to-implement-good-seo-strategy.
+- There are no params in the URL, this can be setup in Google Search Console.			
+
+_Why:_
+> URLs are extremely vital to your SEO campaign. Be sure to use URLs that make it easy for Google™ to index your site. [Read more...](https://support.google.com/webmasters/answer/6080548?visit_id=0-636594791880851880-2150656090&hl=en&rd=2)
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: I'd remove the `tm` from `Google™` to be consistent

File: frontend/docs/seo-guide.md [hunk 17]
```diff
@@ -0,0 +1,393 @@
...
+- Successful migrations to HTTPS are now less prone to lose PageRank.
+- **The best redirect is where every other element stays the same, as much as possible, except for the URL.**
+
+> What means each redirection? [Read more...](https://moz.com/learn/seo/redirection)
+
+#### Robots Exclusion Protocol (REP)
+
+The Robots Exclusion Protocol (REP) is a conglomerate of standards that regulate Web robot behavior and search engine indexing. Despite the "Exclusion" in its name, the REP covers mechanisms for inclusion too.
+
+Some of this mechanisms are: robots.txt, rebots meta elements, microformat relationship-nofollow and sitmaps(We talked about it a few line up). We can divide them in two groups:
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: I'd remove this: `(We talked about it a few line up)`

File: frontend/docs/seo-guide.md [hunk 18]
```diff
@@ -0,0 +1,393 @@
...
+
+_Why:_
+
+> This is a very complex topic, so I recommend you to [read more...](https://developers.google.com/analytics/solutions/)
+
+### 2.6 Social networks
+
+#### Social sharing buttons
+
+Social networks may not play direct role in ranking your site. But social shares generate more eyes focus on your content, and more people you attract, the more likely someone is to link.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: focusing

Comment (SKOLZ) [reply frontend/docs/seo-guide.md]: and the more people you attract

File: frontend/docs/seo-guide.md [hunk 19]
```diff
@@ -0,0 +1,393 @@
...
+#### Social sharing buttons
+
+Social networks may not play direct role in ranking your site. But social shares generate more eyes focus on your content, and more people you attract, the more likely someone is to link. 
+
+_Why:_
+> A study by [BrigthEdge](https://www.brightedge.com/) found that prominent social sharing buttons can increase social sharing by 700%. [Read more...](https://blog.hubspot.com/blog/tabid/6307/bid/24994/using-social-sharing-buttons-leads-to-7x-more-mentions-data.aspx)
+
+#### The Open Graph
+
+The Open Graph protocol enables any web page to become a rich object in a social graph. For instance, this is used on Facebook to allow any web page to have the same functionality as any other object on Facebook.
```
Comment (SKOLZ) [inline frontend/docs/seo-guide.md]: used by Facebook

Comment (SKOLZ) [reply frontend/docs/seo-guide.md]: object in Facebook

Comment (emanuelcasco) [reply frontend/docs/seo-guide.md]: I think the correct way is really  "...object from Facebook".

File: frontend/docs/seo-guide.md (hunk 17 above)
Comment (emanuelcasco) [reply frontend/docs/seo-guide.md]: I do not think so ... It's true we mention it some lines up, but we do not describe it.

File: frontend/docs/seo-guide.md (hunk 15 above)
Comment (emanuelcasco) [reply frontend/docs/seo-guide.md]: I think I fixed the indentation. 
About text meaning, what I tried to explain that URLs must have your best keywords and human-readable text in it. You should avoid ids or not-friendly text in them, like the examples below

File: frontend/docs/seo-guide.md [hunk 20]
```diff
@@ -0,0 +1,393 @@
...
+You can pay for traffic using the PPC advertising programs provided by Google Adwords, Yahoo Search Marketing and others. **This guidelines do not contemplate this techniques**.
+
+
+### 1.2 Index
+
+Google stores all web pages that it knows about in its index. The index entry for each page describes the content and location (URL) of that page. 
+
+A page is indexed by Google if it has been visited by the Google crawler, analyzed for content and meaning, and stored in the Google index. It is important to know that Google does not index all pages but only which ones that seems to be relevent to it. To learn more visit [this site](https://www.google.com/search/howsearchworks/).
+
+To see indexed pages in your site, use the `site:` operator, like this: `site:google.com`. [Read more...](https://support.google.com/webmasters/answer/35256?hl=en)
```
[the remark this answers is not in the document]
Comment (emanuelcasco) [reply frontend/docs/seo-guide.md]: It's strange, that operator allows you to perform a search for a whole domain. So it should open Google SERPs and shows you only results that belongs to `google` domain.

For example (Pay attention to the domain):

- www.google.com
- www.development.google.com
- www.google.com/webmaster

File: frontend/docs/seo-guide.md [hunk 21]
```diff
@@ -0,0 +1,393 @@
...
+- There are not pages without H1.
+
+#### Write valid code
+
+Ensure your code has valid markup. You can use the following tools for that:
+
+- [HTML Validator](https://validator.w3.org/).
+- [CSS Validator](https://jigsaw.w3.org/css-validator/).
+
+Keep it in mind, W3C validation is not the “SEO magic bullet” to top rankings.
```
[the remark this answers is not in the document]
Comment (emanuelcasco) [reply frontend/docs/seo-guide.md]: I rewrite it _"Keep in mind, W3C validation is not the "SEO magic bullet" to achieve high positions. A good HTML syntax does not ensure you a good place in Google SERPs, but a bad syntax move us away them."_.

File: frontend/docs/seo-guide.md [hunk 22]
```diff
@@ -0,0 +1,393 @@
...
+
+#### Headings
+
+>  *“We do use H tags to understand the structure of the text on a page better”* John Mueller, Google 2015
+
+The six heading elements, H1 through H6, denote section headings, with **H1 as the most important and H6 as the least**. The H1 tag is the most important heading because it’s the highest level tag that shows **what your specific page is about**.
+
+Although the order and occurrence of headings is not constrained by the HTML DTD, documents should not skip levels (for example, from H1 to H3), as converting such documents to other representations is often problematic.
+
+You have to keep in mind that headers implementation is one of the more highlighted elements in SEO.
```
[the remark this answers is not in the document]
Comment (emanuelcasco) [reply frontend/docs/seo-guide.md]: My mistake, is "headings implementation..."

**Comment by emanuelcasco** [comment]
Change title to follow the PR naming convention described in #90.

**Comment by mishuagopian** [comment]
👏 @emanuelcasco

File: frontend/docs/seo-guide.md [hunk 23]
```diff
@@ -0,0 +1,394 @@
...
+_Why:_
+> Title tags are a major factor in helping search engines understand what your page is about, and they are the first impression many people have of your page. [Read more...](https://moz.com/learn/seo/title-tag)
+
+#### Images
+
+Almost every content has to have an image in it. But, how this affect SEO? Well, without the proper optimization, however, you’re wasting a valuable SEO asset. Plus, the larger the file size, the longer it takes your page to load, and the higher your bounce rate will be.
+
+- Be sure that your images are no larger than they need to be and that they are in the right file format.
+  _Why:_
+  > Your page's load time will vary based on the number and size of your images.
```
Comment (FrankIglesias) [inline frontend/docs/seo-guide.md]: I'd change images for `assets`  in order to also reffer to videos, fonts etc

File: frontend/docs/seo-guide.md (hunk 9 above)
Comment (FrankIglesias) [reply frontend/docs/seo-guide.md]: :point_up_2:

File: frontend/docs/seo-guide.md (hunk 8 above)
Comment (FrankIglesias) [reply frontend/docs/seo-guide.md]: :point_up_2:

**Pull request merged**